docs/virtualpage.rst

```diff
@@ -0,0 +1,293 @@
+You do not need to fully understand the code to notice that Dynamic Content support requires just
+few lines of extra code.
+
+.. php:method:: getURL($html_wrapping)
+
+    You can use this shortcut method instead of $vp->cb->getURL().
+
+.. php:attr:: $ui
+
+When using 'popup' mode, the output appears inside a `<div class="ui container">`. If you want to change this
+class, you can set $ui property to something else. Try::
+
+    $vp = $layout->add('VirtualPage');
+    $vp->add('LoremIpsum');
+    $vp->ui = 'red inverted segment';
+
+    $label = $layout->add('Label');
+
+    $label->detail = $vp->cb->getURL('popup');
+    $label->link($vp->cb->getURL('popup'));
+
+
+
+
+
+Loader
+------
+
+.. php:class:: Loader
+
+.. php:method:: set()
+
+Loader is designed to delay some slow-loading content by loading it dynamically, after main
+page is rendered.
+
+Comparing to VirtualPage which is a D.Y.I. solution - Loader can be used out of the box.
+Loader extends VirtualPage and is quite similar to it.
+
+Like with a VirtualPage - you should use `set()` to define content that will be loaded dynamically,
+while a spinner is shown to a user::
+
+    $loader = $app->add('Loader');
+    $loader->set(function($p) {
+
+        // Simulate slow-loading component
+        sleep(2);
+        $p->add('LoremIpsum');
+
+    });
+
+A good use-case example would be a dashboard graph. Unlike VirtualPage which is not visible to a regular render,
+Loader needs to occupy some space, so adding other objects inside content can be a good filler until content is loaded::
+
+    $loader = $app->add('Loader');
+    $loader->add(['Message', 'Please wait until we load LoremIpsum...', 'red']);
+    $loader->set(function($p) {
+
+        // Simulate slow-loading component
+        sleep(2);
+        $p->add('LoremIpsum');
+
+    });
+
+
+Triggering Loader
+^^^^^^^^^^^^^^^^^
+
+By default, Loader will display a spinner and will start loading it's contents as soon as DOM Ready() event fires.
+Sometimes you want to control the event. 
+
+.. php:method:: jsLoad()
+
+Returns JS action which will trigger loading. The action will be carried out in 2 steps:
+
+- loading indicator will be displayed
+- JS will request content from $this->getURL() and provided by set()
+- Content will be placed inside Loader's DIV
+- loading indicator will is hidden
+
+.. php:attr:: $loadEvent = null
+
+If you have NOT invoked jsLoad in your code, Loader will automatically assign it do DOM Ready(). If the automatic
+behaviour does not work, you should set value for $loadEvent:
+
+- null = load on DOM ready unless you have invoked jsLoad() in the code.
+- true = load on DOM ready
+- false = never load
+- "string" - bind to custom JS event
+
+To indicate how custom binding works::
+
+    $loader = $app->add(['Loader', 'loadEvent' => 'kaboom']);
+
+    $loader->set(function($p){
+        $p->add('LoremIpsum');
+    });
+
+
+    $app->add(['Button', 'Load data'])->on('click', $loader->js()->trigger('kaboom'));
+
```

Comment:
Not sure what you mean here, this property loadEvent does not exist....

Reply:
yes, i recommend to add it. The property may be "true" by default and it can be passed as 1st argument to js() function inside render().